core/src/main/java/io/questdb/network/AbstractIODispatcher.java

```diff
@@ -107,13 +109,20 @@ public AbstractIODispatcher(
     private void createListenFd() throws NetworkError {
         this.serverFd = nf.socketTcp(false);
         final int backlog = configuration.getListenBacklog();
-        if (nf.bindTcp(this.serverFd, configuration.getBindIPv4Address(), configuration.getBindPort())) {
+        final int bindPort = this.port == 0
```

Comment:
this block looks unnecessarily complex on port logic. `bindPort` variable can be removed and this line can be added:

```java
        if (this.port == 0) {
            this.port = configuration.getBindPort();
        }
```

I think this block deserves explanation (comments).

Additionally:

`createListenFd()` is being called from the constructor and when re-establishing listener after too many connections situation. In the latter port 0 looks like untested, e.g. if the above `if` is removed the listener will resume on new port (potentially different one), but tests pass

Reply:
TL;DR: I think the logic I have should stay, but needs commenting. 

What I implemented is:

Port 0 is resolved only once and then cached to `this.port` for the lifetime of the `IODispatcher` object. 

If the listener is closed and re-established, it will re-use the same port it had used earlier by listening again to `this.port` (and not 0).

Note that the OS reuses a closed port in the ephemeral range only once all other ports in the range are exhausted. In other words, it's pretty unlikely that a port clash would occur in a testcase scenario as the same port would be bound again pretty quickly. 

Were we to use port 0 in production this would be different: We would require different logic and a registrar service for ephemeral questdb instances and we would rebind to a different port 0 each time.

I don't believe we have any use cases for port 0 in production currently, so I think the current logic is fine the way it is as it assures testcases don't go wrong without needing to update port numbers in the Java objects for any fixed-port reconnecting logic algorithms we may have now or in the future.

Reply:
Done.